__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -0,0 +1,1508 @@
+        mocked(vscode.window.showQuickPick).mockResolvedValueOnce("Yes" as any);
+        const deleteSpy = jest.spyOn(blockMocks.mvsApi, "deleteDataSet");
+        deleteSpy.mockClear();
+
+        await expect(dsActions.deleteDataset(child, blockMocks.testDatasetTree)).rejects.toEqual(Error("deleteDataSet() called from invalid node."));
+        expect(deleteSpy).not.toBeCalled();
+    });
+});
+
+describe("Dataset Actions Unit Tests - Function enterPattern", () => {
+    let blockMocks;
+
+    function createBlockMocks() {
+        const session = generateISession();
+        const imperativeProfile = generateIProfile();
+        const zosmfSession = generateBasicZosmfSession(imperativeProfile);
+        const treeView = generateTreeView();
+        const datasetSessionNode = generateDatasetSessionNode(session, imperativeProfile);
+        const mvsApi = generateMvsApi(imperativeProfile);
+        bindMvsApi(mvsApi);
+
+        return {
+            session,
+            zosmfSession,
+            treeView,
+            imperativeProfile,
+            datasetSessionNode,
+            mvsApi,
+            testDatasetTree: generateDatasetTree(datasetSessionNode, treeView)
+        };
+    }
+
+    afterAll(() => jest.restoreAllMocks());
+
+    it("Checking common dataset filter action", async () => {
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const node = new ZoweDatasetNode("node", vscode.TreeItemCollapsibleState.None, blockMocks.datasetSessionNode, null);
+        node.pattern = "TEST";
+        node.contextValue = globals.DS_SESSION_CONTEXT;
+
+        mocked(vscode.window.showInputBox).mockResolvedValueOnce("test");
+        await dsActions.enterPattern(node, blockMocks.testDatasetTree);
+
+        expect(mocked(vscode.window.showInputBox)).toBeCalledWith({
+            prompt: "Search data sets by entering patterns: use a comma to separate multiple patterns",
+            value: node.pattern
+        });
+        expect(mocked(vscode.window.showInformationMessage)).not.toBeCalled();
+    });
+    it("Checking common dataset filter failed attempt", async () => {
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const node = new ZoweDatasetNode("node", vscode.TreeItemCollapsibleState.None, blockMocks.datasetSessionNode, null);
+        node.pattern = "TEST";
+        node.contextValue = globals.DS_SESSION_CONTEXT;
+
+        mocked(vscode.window.showInputBox).mockResolvedValueOnce("");
+        await dsActions.enterPattern(node, blockMocks.testDatasetTree);
+
+        expect(mocked(vscode.window.showInformationMessage)).toBeCalledWith("You must enter a pattern.");
+    });
+    it("Checking favorite dataset filter action", async () => {
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const favoriteSample = new ZoweDatasetNode("[sestest]: HLQ.TEST", vscode.TreeItemCollapsibleState.None, undefined, null);
```

Comment:
Reusable node, define in createBlockMocks(), and then reassign only in this it() block?